pkg/client/factory.go

```diff
@@ -108,18 +147,38 @@ func (f *factory) KubeClient() (kubernetes.Interface, error) {
 }
 
 func (f *factory) DynamicClient() (dynamic.Interface, error) {
-	clientConfig, err := Config(f.kubeconfig, f.kubecontext, f.baseName)
+	clientConfig, err := f.ClientConfig()
 	if err != nil {
 		return nil, err
 	}
-
 	dynamicClient, err := dynamic.NewForConfig(clientConfig)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 	return dynamicClient, nil
 }
 
+func (f *factory) WithBasename(name string) Factory {
```

Comment:
Does this need to return a new `Factory`? I'd think this could be `SetBasename` and just update the field, similarly to the methods for QPS and burst, right?

Reply:
I didn't want to update the basename on the client that's used for anything besides the `server` command. However, now that I'm thinking more on it, it would almost always be a singleton since we can't invoke another command in the exact same binary. I'll update that.